### ISSUE TITLE
[FEAT] Review Flashcards from All Decks Simultaneously

### ISSUE THREAD

**Issue by 23727** [st3v3nmw/obsidian-spaced-repetition#415]
**Is your feature request related to a problem? Please describe.**
At the moment, unless all cards are under a parent deck, we can't review all the cards in one session

**Describe the solution you'd like**
I would like an option to study all cards across all decks
It would be nice to have the option to study these cards in random order

Thank you

**Comment by AB1908** [comment]
Wait, isn't this default behavior? If you tag the note with `#flashcards` all of them show up in a single deck for me.

**Comment by 23727** [comment]
> Wait, isn't this default behavior? If you tag the note with `#flashcards` all of them show up in a single deck for me.

Yes, it can be when using the tagging option for deck creation. 

With the option to turn folders into decks, we're left with top level decks (which, are in the root folder of our vault), such as the following

![image](https://user-images.githubusercontent.com/84650530/157506080-db911204-ebd0-4ab6-b245-fe2b4b25ae95.png)

I just started using the spaced repetition plug-in again, so there isn't many decks to see; however, I hope it is visible that there is no option to review all cards when using folders/subfolders as decks and I think it would a really nice feature to have.

Thank you

**Comment by AB1908** [comment]
Ah, I see. Makes sense. I'll see what I can find.

**Comment by 23727** [comment]
My current workaround it to put all folders into a 'Home' folder that is in the root of the Obsidian vault

**Comment by AB1908** [comment]
I haven't had the time to dig into the deck construction but I'll keep an eye out to see if it clicks for me.

**Comment by ronzulu** [comment]
Hi @cesar-programs 

I'm sure you have completely forgotten about your idea :-) but this has now been implemented, and is ready for beta testing.

If there are multiple top-level decks, then the flashcard modal displays an entry for `All Decks ` for example:

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/b1874625-4d54-4976-a7a1-0aeef3e83fd4)

The latest mainstream release 1.11.0 also includes the ability to shown random cards across all decks. This is a new setting:

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/7f269144-d6f1-4eff-b1ed-13a6fef1b2ec)

# Beta Testing

There is a file called `main.js` in this zip file
[main_415A.zip](https://github.com/st3v3nmw/obsidian-spaced-repetition/files/13927264/main_415A.zip)

If you wish to try this before it is included in a mainstream release,  you will need to copy that file to the obsidian plug-in folder, e.g.
`D:\Obsidian\Obsidian\.obsidian\plugins\obsidian-spaced-repetition`

Then reload the plug-in, or restart obsidian. 

Any feedback appreciated.

Cheers
Ronny

**Comment by AB1908** [comment]
@ronzulu you've probably seen it already but if not, can you look into using tfthacker's brat tool for easier beta distribution?

**Comment by ronzulu** [comment]
> @ronzulu you've probably seen it already but if not, can you look into using tfthacker's brat tool for easier beta distribution?

Thanks for letting me know, I hadn't come across it before.

**Comment by KyleKlus** [comment]
HI, this one is implemented now😇
